src/components/Modal.jsx

```diff
@@ -0,0 +1,50 @@
+import ReactDom from 'react-dom';
+import styled from 'styled-components';
+
+const Dimmer = styled.div`
+  width: 100%;
+  height: 100%;
+
+  display: flex;
+  flex-direction: column;
+  justify-content: flex-end;
+  align-items: center;
+
+  position: absolute;
+  inset: 0;
+
+  background: rgba(0, 0, 0, 0.5);
+`;
+
+const ModalBox = styled.div`
+  width: 25%;
+  height: auto;
+  border-radius: 10px;
+  background: white;
+
+  display: flex;
+  flex-direction: column;
+  gap: 30px;
+
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+  opacity: 0.9;
+
+  padding: 20px;
+`;
+
+const Modal = ({ toggleModal, children }) => {
+  const modalElement = document.getElementById('modal');
+
+  return ReactDom.createPortal(
+    <div>
+      <Dimmer onClick={toggleModal} />
```

Comment:
`Dimmer`는 항상 모달이 열린 상태에서만 보이게 될 테니, Dimmer를 클릭하면 모달을 '토글'하기보다는 항상 모달을 '닫는다'가 더 자연스럽진 않을까요? 
```suggestion
      <Dimmer onClick={closeModal} />
```